tripy/tripy/frontend/trace/ops/stack.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Union
+from tripy import export, constraints
+from tripy.common.exception import raise_error
+
+
+@export.public_api(document_under="operations/functions")
+@constraints.dtype_info(
+    dtype_variables={
+        "T1": ["float32", "float16", "bfloat16", "float8", "int8", "int32", "int64", "bool"],
+    },
+    dtype_constraints={"tensors": "T1", constraints.RETURN_VALUE: "T1"},
+)
+def stack(tensors: List[Union["tripy.Tensor"]], dim: int = 0) -> "tripy.Tensor":
+    """
+    Stacks multiple tensors of same shape along a given dimension.
```

Comment:
Same shape except along the concat axis right?

Reply:
No, stack always appends a dimension along which concatenation happens.